File: 2023-2024/sourcecode/mod3/RetrofitBasicsJava/app/src/main/java/com/example/retrofitbasicsjava/ui/PostComparator.java
package com.example.retrofitbasicsjava.ui;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.DiffUtil;

import com.example.retrofitbasicsjava.data.Post;

public class PostComparator extends DiffUtil.ItemCallback<Post> {
    @Override
    public boolean areItemsTheSame(@NonNull Post oldItem, @NonNull Post newItem) {
        return oldItem == newItem;
    }

    @Override
    public boolean areContentsTheSame(Post oldItem, @NonNull Post newItem) {
        return oldItem.equals(newItem);
    }
}

File: 2023-2024/sourcecode/mod4/lab12/RetrofitBasicsKotlin/app/src/main/java/com/example/retrofitbasicskotlin/ui/PostComparator.kt
package com.example.retrofitbasicskotlin.ui

import androidx.recyclerview.widget.DiffUtil
import com.example.retrofitbasicskotlin.data.Post

class PostComparator : DiffUtil.ItemCallback<Post>() {
    override fun areItemsTheSame(oldItem: Post, newItem: Post): Boolean {
        return oldItem === newItem
    }

    override fun areContentsTheSame(oldItem: Post, newItem: Post): Boolean {
        return oldItem == newItem
    }
}